tests/rptest/tests/upgrade_test.py

```diff
@@ -506,7 +506,9 @@ def verify():
                 'cloud_storage_manifest_max_upload_interval_sec':
                 1,
                 'cloud_storage_spillover_manifest_max_segments':
-                2
+                2,
+                'cloud_storage_spillover_manifest_size':
```

Comment:
Worth a comment. I don't understand why this is here.

Reply:
Disable spillover? Why would we have `cloud_storage_spillover_manifest_max_segments` then?

Reply:
the jist is that *size takes precedence over the *max_segments variant https://github.com/redpanda-data/redpanda/blob/cc6bae525e5acdc24397f266876e2d5511a9fe3a/src/v/archival/ntp_archiver_service.cc#L2471-L2483 
Now that the *size variant has a default value, the tests failed because they were expecting a fast spillover while Redpanda behaves in a more "production mode" manner.
btw the first commit is about reverting a fix (setting *size to None) done in a central place because it didn't catch all the uses of *max_segments.

Reply:
i'll push another commit with comments later after this ci run, don't want to lose the findings if other random tests have some dependency on fast spillovers